File: library/src/main/java/com/thesurix/gesturerecycler/DefaultItemClickListener.java
package com.thesurix.gesturerecycler;

/**
 * Default implementation of the {@link RecyclerItemTouchListener.ItemClickListener}.
 * @author thesurix
 */
public class DefaultItemClickListener<T> implements RecyclerItemTouchListener.ItemClickListener<T> {

    @Override
    public boolean onItemClick(final T item, final int position) {
        return false;
    }

    @Override
    public void onItemLongPress(final T item, final int position) {

    }

    @Override
    public boolean onDoubleTap(final T item, final int position) {
        return false;
    }
}

File: library/src/main/java/com/thesurix/gesturerecycler/DefaultItemClickListener.kt
package com.thesurix.gesturerecycler

/**
 * Default implementation of the [RecyclerItemTouchListener.ItemClickListener].
 * @author thesurix
 */
open class DefaultItemClickListener<T> : RecyclerItemTouchListener.ItemClickListener<T> {

    override fun onItemClick(item: T, position: Int): Boolean {
        return false
    }

    override fun onItemLongPress(item: T, position: Int) {
    }

    override fun onDoubleTap(item: T, position: Int): Boolean {
        return false
    }
}
